inst/extdata/install-packages.R

```diff
@@ -0,0 +1,163 @@
+#Download paths of latest develop package versions
+.developPaths <- list(ospsuite.utils = "https://ci.appveyor.com/api/projects/open-systems-pharmacology-ci/ospsuite-rutils/artifacts/ospsuite.utils.zip",
+                      tlf = "https://ci.appveyor.com/api/projects/open-systems-pharmacology-ci/tlf-library/artifacts/tlf.zip",
+                      ospsuite = "https://ci.appveyor.com/api/projects/open-systems-pharmacology-ci/ospsuite-r/artifacts/ospsuite.zip")
+
+#' Test connection to PK-Sim
+#'
+#' @param pkSimPath Path where PK-Sim is installed. If this is not specified
+#' (`NULL`), path is estimated by the `ospsuite` package.
+#'
+#' @return `TRUE` if `ospsuite::initPKSim(pkSimPath)` is successful.
+testPKSIMConnection <- function(pkSimPath = NULL) {
+  # returns TRUE if PK-Sim.R dll can be loaded,
+  # or raises errors otherwise
+  library(ospsuite)
+  ospsuite::initPKSim(pkSimPath)
+  return(TRUE)
+}
+
+# Remove functions created by this script from the environment
+cleanEnvironment <- function(){
+  rm(packageInstallationMessages,
+          testPKSIMConnection,
+          cleanEnvironment,
+          displayProgress,
+          installOSPPackages,
+          installPackagesGlobally
+  )
+}
+
+displayProgress <- function(current, success = TRUE, message = NULL, suppressOutput = TRUE) {
+  states <- c("Installing RENV", "Installing CRAN packages",
+              "Checking RTOOLS", "Installing rClr", "Installing ospsuite.utils",
+              "Installing tlf", "Installing ospsuite",
+              "Testing PK-Sim connection",
+              "Installation successful")
+  if (suppressOutput) {
+    cat("\014")
+    for (i in seq_along(states)) {
+      cat(paste0(ifelse(i < which(states == current), "V", ifelse(i > which(states == current), ".", ifelse(success == FALSE, "X", ".")))), " ", states[[i]], "\n")
+    }
+  } else {
+    cat(paste0(current, "\n"))
+  }
+  if (!is.null(message)) {
+    message(message)
+  }
+}
+
+#' Install all osps packages and their dependencies.
+#'
+#' @param rtoolsPath Path to where rtools are installed. If `NULL` (default),
+#' the path is deduced from system environment variables.
+#' @param rclrVersion Version of rClr package. Default is 0.9.2 for Windows R4.
+#' @param developerVersion If `FALSE` (default), release verions of the packages
+#' will be installed. If `TRUE`, latest developer builds of the osps packages
+#' will be installed
+#' @param suppressOutput
+#'
+#' @return
+#' @export
+#'
+#' @examples
+installOSPPackages <- function(rtoolsPath = NULL, rclrVersion = "0.9.2",
+                               suppressOutput = TRUE,
+                               developerVersion = FALSE){
+  # Install dependencies from CRAN
+  displayProgress("Installing CRAN packages", suppressOutput = suppressOutput)
+  install.packages(c("R6", "stringr", "readr", "hash", "readxl", "shiny", "shinyjs", "vctrs", "writexl", "dplyr", "tidyr", "ggplot2", "FME", "patchwork"),
```

Comment:
Installing CRAN packages

Reply:
I miss the following packages:
* **purrr**
* **jsonlite**
* **knitr**
* **rmarkdown**

On the other side, why are the packages bellow installed? None of them is in the dependencies list of ospsuite.utils, tlf or ospsuite-R
* **hash**
* **shinyjs**
* **vctrs**
* **writexl**
* **FME**

Reply:
`hash` and `FME` are (currently) required for parameter identification

Reply:
ok, but parameter identification is standalone package and not (yet) part of the ospsuite package.
Would suggest to remove them.

Reply:
- purrr is probably some dependency from other package....
- jsonlite is not a dependency of ospsuite
- **knitr** and **markdown** are only required for building vignettes but not for simply install and work with the package
- I removed the unnecessary packages from installation

Reply:
let's install **purrr** explicitely. Otherwise if it will be not a dependnecy of other CRAN packages one day - the installation script will become invalid.
For the other packages - agree. They are used by the `ospsuite.reportingengine` and must be installed there, not as part of ospsuite-R.